celery_utils/backends.py

```diff
@@ -0,0 +1,70 @@
+"""
+Custom backends to be used as CELERY_RESULT_BACKEND.
+"""
+from datetime import datetime, timedelta
+import json
+
+from djcelery.backends.database import DatabaseBackend
+from djcelery.models import TaskMeta
+
+from models import ChordData
+
+
+class ChordableDjangoBackend(DatabaseBackend):
+    """
+    A django backend that is friendly to chord usage.
+    """
+
+    EXPIRES_AFTER = timedelta(days=7)
+
+    def cleanup(self):
+        chords_to_delete = ChordData.objects.filter(
+            callback_result__date_done__lte=datetime.now() - self.EXPIRES_AFTER
+        ).iterator()
+        for chord in chords_to_delete:
+            subtask_ids = [subtask.task_id for subtask in chord.completed_results.all()]
+            chord.completed_results.clear()
+            TaskMeta.objects.filter(task_id__in=subtask_ids).delete()
+            chord.callback_result.delete()
+            chord.delete()
+
+    def add_to_chord(self, chord_id, result):
+        pass
+
+    def on_chord_part_return(self, task, state, result, propagate=None):
+        """
+        Need to dig a bit more on the specific values of params, but this is where we
+        should check to see if callback should be fired and fire it.
+        """
+        subtask = TaskMeta.objects.update_or_create(task_id=task.request.id, defaults={'status': state, 'result': result})
+        chord_data = ChordData.objects.get(callback_result__task_id=task.request.chord['options']['task_id'])
+        if chord_data.is_ready():
+            chord_data.execute_callback()
+
+    # I think there's some sort of on_chord_error_return handler, will look further after proof-of-concept is working
```

Comment:
This has been resolved in the latest commit; I let the subtasks deal with their own retries under-the-hood, and implemented the skip-or-fail logic in `execute_callback()`.